test/format/format_prepare_discover.c

```diff
@@ -0,0 +1,123 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "format.h"
+
+/*
+ * wts_prepare_discover --
+ *     Discover and process prepared transactions.
+ */
+void
+wts_prepare_discover(WT_CONNECTION *conn)
+{
+    /* Claim prepare is not supported in in-memory runs. */
+    if (GV(RUNS_IN_MEMORY))
+        return;
+
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_SESSION *session;
+    uint64_t prepared_id, ts;
+    uint32_t claim_count, discover_count, rand_val;
+    char buf[128];
+    bool should_commit;
+    /*
+     * Individual object verification. Do a full checkpoint to reduce the possibility of returning
+     * EBUSY from the following verify calls.
+     */
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Open the prepare discover cursor */
+    ret = session->open_cursor(session, "prepared_discover:", NULL, NULL, &cursor);
+    if (ret == WT_NOTFOUND) {
+        /* No prepared transactions found - this is normal */
+        testutil_check(session->close(session, NULL));
+        return;
+    }
+    testutil_check(ret);
+
+    /*
+     * If the cursor was opened successfully, there should be at least one prepared transaction to
+     * discover.
+     */
+    trace_msg(session, "Starting prepare discover operation %s", "");
```

Comment:
Sorry don't know much about race_msg but does this require creating a new trace flag? Why use the default